tests/executors/test_celery_executor.py

```diff
@@ -225,19 +225,19 @@ def test_retry_on_error_sending_task(self, caplog):
 
             # Test that when heartbeat is called again, task is published again to Celery Queue
             executor.heartbeat()
-            assert dict(executor.task_publish_retries) == {key: 2}
+            assert dict(executor.task_publish_retries) == {key: 1}
```

Comment:
Why did these numbers change? Ideally this refactor shouldn't have changed anything about the behaviour, no?

Reply:
I think that is because now we use "Counter" from collections that starts with "0", whereas previously we started with "1" with OrderedDict

Reply:
and before the condition was

https://github.com/apache/airflow/blob/d4a33480550db841657b998c0b4464feffec0ef9/airflow/executors/celery_executor.py#L288-L290

and now it is 
https://github.com/apache/airflow/blob/5b5511e6e095e68d41a0713aa563772fe847a4c8/airflow/executors/celery_executor.py#L269

Reply:
Exactly – and this is definitely a bit of a tricky change, but I do think it makes sense because the code is just easier with the new indexing logic.